pkg/workloadmeta/store.go

```diff
@@ -30,6 +30,50 @@ const (
 	eventChBufferSize      = 50
 )
 
+// Store is the central storage of metadata about workloads.
+//
+// Typically there is only one instance, accessed via GetGlobalStore.
+type Store interface {
+	// Start starts the store, asynchronously initializing collectors and
+	// beginning to gather workload data.  This is typically called during
+	// agent startup.
+	Start(ctx context.Context)
+
+	// Subscribe subscribes the caller to events representing changes to the
+	// store, limited to events matching the filter.  The name is used for
+	// telemetry and debugging.
+	//
+	// The first message on the channel is special: it contains an EventTypeSet
+	// event for each entity currently in the store.  If the Subscribe call
+	// occurs during agent startup, then it is safe to assume that entities
+	// referenced in the first message were running before the agent started.
```

Comment:
Might be worth point it out that this is inherently racy, so assuming that anything NOT in the first message was NOT running before the agent started could very well be wrong ;)

Reply:
Good point, thanks.